Objects/Enemy/Enemy.gd

```diff
@@ -27,29 +37,38 @@ func ai_move_toward(_target, delta):
 	else:
 		velocity *= speed
 
-	var collision = move_and_collide(velocity * delta)
-	if collision:
-		# slide reaction
-		velocity = velocity.slide(collision.normal)
+	# move
+	move_and_slide(velocity)
 
-		# detect collision object (vehicle? player?)
-		if collision.get_collider() == Vehicle:
+	# detect collision
+	if ray.is_colliding():
+		if ray.get_collider() == Vehicle:
```

Comment:
While this works, one way you could implement this is to do a group check on the collider. For ex:
`if ray.get_collider().is_in_groups("group name")`.

That way, you don't have to store a reference to a node on ready _and_ the enemy does not have to assume that a vehicle and player exists as siblings on the parent node.